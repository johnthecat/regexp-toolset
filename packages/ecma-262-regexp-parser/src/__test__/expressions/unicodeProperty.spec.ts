import { describe, it, expect } from 'vitest';
import { parseRegexp } from '../../index.js';

/**
 * @see parseUnicodeProperty
 */
describe.each(['L', 'Hex', 'General_Category=Letter', 'Script=Latin'])('Unicode property %o', unicodeProperty => {
  it('should parse unicode property', () => {
    expect(parseRegexp(`/\\p{${unicodeProperty}}/u`)).toMatchSnapshot();
  });
  it('should parse non unicode property', () => {
    expect(parseRegexp(`/\\P{${unicodeProperty}}/u`)).toMatchSnapshot();
  });
});
