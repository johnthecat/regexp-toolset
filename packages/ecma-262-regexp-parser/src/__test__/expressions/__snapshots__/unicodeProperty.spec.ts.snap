// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unicode property 'General_Category=Letter' > should parse non unicode property 1`] = `
{
  "body": {
    "end": 27,
    "kind": "NonUnicodeProperty",
    "name": "General_Category",
    "start": 1,
    "value": "Letter",
  },
  "end": 29,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'General_Category=Letter' > should parse unicode property 1`] = `
{
  "body": {
    "end": 27,
    "kind": "UnicodeProperty",
    "name": "General_Category",
    "start": 1,
    "value": "Letter",
  },
  "end": 29,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'Hex' > should parse non unicode property 1`] = `
{
  "body": {
    "end": 7,
    "kind": "NonUnicodeProperty",
    "name": "Hex",
    "start": 1,
    "value": null,
  },
  "end": 9,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'Hex' > should parse unicode property 1`] = `
{
  "body": {
    "end": 7,
    "kind": "UnicodeProperty",
    "name": "Hex",
    "start": 1,
    "value": null,
  },
  "end": 9,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'L' > should parse non unicode property 1`] = `
{
  "body": {
    "end": 5,
    "kind": "NonUnicodeProperty",
    "name": "L",
    "start": 1,
    "value": null,
  },
  "end": 7,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'L' > should parse unicode property 1`] = `
{
  "body": {
    "end": 5,
    "kind": "UnicodeProperty",
    "name": "L",
    "start": 1,
    "value": null,
  },
  "end": 7,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'Script=Latin' > should parse non unicode property 1`] = `
{
  "body": {
    "end": 16,
    "kind": "NonUnicodeProperty",
    "name": "Script",
    "start": 1,
    "value": "Latin",
  },
  "end": 18,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Unicode property 'Script=Latin' > should parse unicode property 1`] = `
{
  "body": {
    "end": 16,
    "kind": "UnicodeProperty",
    "name": "Script",
    "start": 1,
    "value": "Latin",
  },
  "end": 18,
  "flags": "u",
  "kind": "Regexp",
  "start": 0,
}
`;
